neuralop/training/training_state.py

```diff
@@ -24,12 +25,16 @@ def load_training_state(save_dir: Union[str, Path], save_name: str,
     save_name : str
         name of model to load
     """
+    map_location = None
+    if dist.is_initialized():
+        map_location = {"cuda:0" : f"cuda:{dist.get_rank}"}
```

Comment:
I would let the user specify an optional map_location. This may not always work.